sklearn/ensemble/tests/test_forest.py

```diff
@@ -1854,3 +1860,70 @@ def test_non_supported_criterion_raises_error_with_missing_values():
     msg = "RandomForestRegressor does not accept missing values"
     with pytest.raises(ValueError, match=msg):
         forest.fit(X, y)
+
+
+@skip_if_no_network
```

Comment:
This require to fetch Ames housing. I don't know if we could kind of cache the dataset to make sure that we run the test instead of skipping it.

Reply:
I would very much rather have a toy dataset made for this test rather than downloading one.

Reply:
Although this test is useful, I prefer to hold off on adding it because running `fetch_openml` can lead to rare conditions when running with `pytest-xdist`. There is no race condition now because `fetch_openml(("house_prices")`, but I rather not establish the pattern.

Note, much of https://github.com/scikit-learn/scikit-learn/blob/cb836be0ff8347ccb0ab722760df68d07485101e/sklearn/conftest.py#L66 was written to "download all the datasets before pytest runs".

For this specific PR, I think `test_regression_tree_missing_values_toy` is sufficient as a non-regression test.